File: GraphCodeBert/examples/trial_sample.py
# trial_sample.py

import socket

# 1. Resource leak
def fetch_api(endpoint):
    sock = socket.socket()
    sock.connect(('api.com', 80))
    sock.send(endpoint.encode())
    # Missing sock.close()
    return sock.recv(1024)

# 2. Null pointer dereference (None usage)
def process_data(data):
    if data is None:
        return data.upper()  # Will throw AttributeError

# 3. Concurrency issue
import threading

shared_counter = 0

def increment():
    global shared_counter
    for _ in range(1000):
        shared_counter += 1  # Not thread-safe

t1 = threading.Thread(target=increment)
t2 = threading.Thread(target=increment)
# Missing t1.start() and t2.start() join for proper synchronization

# 4. Security vulnerability
import os

def read_secret_file(filename):
    return os.system("cat " + filename)  # Vulnerable to injection

# 5. Code complexity
def nested_loops():
    for i in range(5):
        for j in range(5):
            for k in range(5):
                print(i, j, k)

File: GraphCodeBert/examples/TrialSample.java
// TrialSample.java

import java.io.File;
import java.io.FileReader;
import java.io.IOException;

// 1. Resource leak
public class TrialSample {
    public String readFile(String filename) throws IOException {
        FileReader fr = new FileReader(filename);
        // Missing fr.close()
        char[] data = new char[100];
        fr.read(data);
        return new String(data);
    }

    // 2. Null pointer dereference
    public void processString(String s) {
        if (s == null) {
            System.out.println(s.length()); // Will throw NullPointerException
        }
    }

    // 3. Concurrency issue
    private int counter = 0;
    public void incrementCounter() {
        counter++; // Not synchronized
    }

    // 4. Security vulnerability
    public void runCommand(String cmd) throws IOException {
        Runtime.getRuntime().exec(cmd); // Vulnerable to injection
    }

    // 5. Code complexity
    public void nestedLoops() {
        for(int i=0; i<5; i++) {
            for(int j=0; j<5; j++) {
                for(int k=0; k<5; k++) {
                    System.out.println(i + " " + j + " " + k);
                }
            }
        }
    }
}